### PR TITLE
fix: add build partitioning

### PR DESCRIPTION
**Pull request by kdoroszko-splunk**
This PR resumes partioned builds creation. 
Jira: https://splunk.atlassian.net/browse/ADDON-72517 
Test run: https://github.com/splunk/splunk-add-on-for-mysql/actions/runs/9890730582 (the partial packages are contained in `package-deployment` artifact)

### CODE REVIEW

**Comment by mkolasinski-splunk** [comment]
@kdoroszko-splunk can you make a beta release on test-addonfactory repo?

File: .github/workflows/reusable-build-test-release.yml [hunk 1]
```diff
@@ -2096,6 +2111,13 @@ jobs:
           git_committer_email: ${{ secrets.SA_GH_USER_EMAIL }}
           gpg_private_key: ${{ secrets.SA_GPG_PRIVATE_KEY }}
           passphrase: ${{ secrets.SA_GPG_PASSPHRASE }}
+      - name: Download package-deployment
+        if: ${{ steps.semantic.outputs.new_release_published == 'true' }}
+        uses: actions/download-artifact@v3
```
Comment (artemrys) [inline .github/workflows/reusable-build-test-release.yml:2116]:
```suggestion
        uses: actions/download-artifact@v4
```

Comment (kdoroszko-splunk) [reply .github/workflows/reusable-build-test-release.yml]: done.

**Review by artemrys — APPROVED** [review]

**Pull request merged**